src/errors.js

```diff
@@ -22,18 +22,27 @@ export type ErrorCode = "PP0001";
 // This is the error format used to report errors to the caller-supplied
 // error-handler
 export class CompilerDiagnostic extends Error {
-  constructor(message: string, location: ?BabelNodeSourceLocation, errorCode: string, severity: Severity) {
+  constructor(
+    message: string,
+    location: ?BabelNodeSourceLocation,
+    errorCode: string,
+    severity: Severity,
+    sourceFilePaths?: { sourceMaps: Array<string>, sourceFiles: Array<{ absolute: string, relative: string }> }
+  ) {
     super(message);
 
     this.location = location;
     this.severity = severity;
     this.errorCode = errorCode;
+    this.sourceFilePaths = sourceFilePaths;
   }
 
   callStack: void | string;
   location: ?BabelNodeSourceLocation;
   severity: Severity;
   errorCode: string;
+  // For --reproOnFatalError, we need to pass the names of all sourcefiles touched by Prepack back to the CLI.
```

Comment:
Is this only used by `reproOnFatalError` or both repro options?

Reply:
The list of files is passed in both flags, but the FatalError flag is more pertinent since this is the error object. I will clarify the comment though